### ISSUE TITLE
micro crashes

### ISSUE THREAD

**Issue by DUOLabs333** [FEX-Emu/FEX#1330]
Micro, a text editor written in Go, crashes.

**Comment by Sonicadvance1** [comment]
Took a peak at this.
First thing we need to support is golang's asynchronous preemption. This can be disabled with `GODEBUG=asyncpreemptoff=1` but it doesn't workaround the Micro issue.

It then throws an assert in a lockfree stack implementation.
```
runtime: lfstack.push invalid packing: node=0xffff842b5b40 cnt=0x1 packed=0xffff842b5b400001 -> node=0xffffffff842b5b40
fatal error: lfstack.push

runtime stack:
runtime.throw(0xa66512, 0xc)
        runtime/panic.go:1117 +0x72
runtime.(*lfstack).push(0xf46288, 0xffff842b5b40)
        runtime/lfstack.go:30 +0x14f
runtime.(*spanSetBlockAlloc).free(...)
        runtime/mspanset.go:292
runtime.(*spanSet).reset(0xf3f890)
        runtime/mspanset.go:265 +0x88
runtime.finishsweep_m()
        runtime/mgcsweep.go:145 +0x9c
runtime.gcStart.func1()
        runtime/mgc.go:1381 +0x25
runtime.systemstack(0x0)
        runtime/asm_amd64.s:379 +0x66
runtime.mstart()
        runtime/proc.go:1246
```

**Comment by Sonicadvance1** [comment]
Looks like the lfstack panics in my case because of a 48bit VA on my device.
This is a difference on Arm64 versus x86-64 where pointers on x86-64 are expected to sext to 48bit->64bit, while on Arm64 it is expected to zext.
If I steal the top 128TB of address space on Arm64 (Which more closely matches what x86-64 behaviour does) then it works with asyncpreempt disabled.
This will need some more investigation to see what a correct workaround is.

**Comment by Sonicadvance1** [comment]
https://github.com/golang/go/issues/49405
Bug reported to golang here.

**Comment by DUOLabs333** [comment]
The micro bug is solved with #1346
